### ISSUE TITLE
Tusky crashes upon proximity of a post

### ISSUE THREAD

**Issue by tleydxdy** [tuskyapp/Tusky#938]
when ever [this](https://pl.smuglo.li/notice/8043569) post is shown, tusky crashes. 

* * * *
- Tusky Version:4.0 fdroid
- Android Version:7.1.2
- Android Device:sony xperia z5
- Mastodon instance (if applicable):gnu/smug

- [✓] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by tleydxdy** [comment]
seems like it was caused by the huge picture.
logcat shows

11-29 12:55:18.846 E/AndroidRuntime(27720): java.lang.RuntimeException: Canvas: trying to draw too large(108000000bytes) bitmap.

**Comment by tleydxdy** [comment]
the picture is 6000x4500 pixels big

**Comment by connyduck** [comment]
Looks like pleroma (or some instances?) don't resize images for preview, so Tusky loads the fullsize images. Thats mostly a pleroma problem, but we still should not crash

**Comment by stom79** [comment]
I got the same issue with Pleroma: https://gitlab.com/tom79/mastalab/issues/521

I now use this helper function (can be improved/customized)
```Java
public static Bitmap compressImageIfNeeded(Bitmap bmToCompress){

        int size = bmToCompress.getByteCount();
        double resizeby =  33554432; //4Mo
        double resize = ((double)size)
        if( resize > 1 ) {
            Bitmap newBitmap = Bitmap.createScaledBitmap(bmToCompress, (int) (bmToCompress.getWidth() / resize),
                    (int) (bmToCompress.getHeight() / resize), true);
            ByteArrayOutputStream bos = new ByteArrayOutputStream();
            newBitmap.compress(Bitmap.CompressFormat.JPEG, 80, bos);
            return newBitmap;
        }
        return bmToCompress;
    }
```

**Comment by connyduck** [comment]
should be fixed by #1043
